Lightyear/Core.idr

```diff
@@ -129,34 +180,55 @@ commitTo (PT f) = PT $ \r, us, cs, ue, ce => f r cs cs ce ce
 interface Stream tok str | str where
   uncons : str -> Maybe (tok, str)
 
+  updatePos : Nat -> Position -> tok -> Pair Position Position
```

Comment:
Thinking about it, update position is required by stream itself to advance the position respective to the stream's implementation.